core/store/ledgerstore/event_store.go

```diff
@@ -212,15 +212,56 @@ func (this *EventStore) getEventNotifyByTxKey(txHash common.Uint256) []byte {
 	return key
 }
 
-func (this *EventStore) SaveContractMetaDataEvent(height uint32, metaData *payload.MetaDataCode) {
+func (this *EventStore) SaveContractMetaDataEvent(height uint32, metaData *payload.MetaDataCode) error {
+	heightsList, err := this.GetContractMetaHeights(metaData.Contract)
+	if err != nil && err != scom.ErrNotFound {
+		return fmt.Errorf("SaveContractMetaDataEvent: get contract meta heights failed, err: %s", err)
+	}
+	heightsNum := len(heightsList)
+	if heightsNum > 0 {
+		if height < heightsList[heightsNum-1] {
+			return fmt.Errorf("SaveContractMetaDataEvent: save height unmatch")
+		} else if height > heightsList[heightsNum-1] {
+			heightsList = append(heightsList, height)
+		}
+	} else {
+		heightsList = []uint32{height}
+	}
+	this.SaveContractMetaHeights(metaData.Contract, heightsList)
 	key := getContractMetaDataKey(height, metaData.Contract)
 	value := common.NewZeroCopySink(64)
 	metaData.Serialization(value)
 	this.store.BatchPut(key, value.Bytes())
+	return nil
 }
 
 func (this *EventStore) GetContractMetaDataEvent(height uint32, contractAddr common.Address) (*payload.MetaDataCode, error) {
-	key := getContractMetaDataKey(height, contractAddr)
+	heightsList, err := this.GetContractMetaHeights(contractAddr)
+	if err != nil {
```

Comment:
how is ErrNotFound processed?

Reply:
no processed, see it as a normal error since reading meta data require the contract must exist.